### ISSUE TITLE
memory

### ISSUE THREAD

**Issue by H-CODE6** [himashi92/VT-UNet#13]
Hello, the device around me now has only 12g video memory. Your code has been reporting video memory errors. Can you run it by modifying parameters?

**Comment by H-CODE6** [comment]
Is there a lighter version that can be reproduced in 12g video memory?

**Comment by himashi92** [comment]
try using embed_dims = 48, which is tiny version of vt-unet.

**Comment by H-CODE6** [comment]
OK, thank you very much for your reply. Is there a big difference in the performance of the three versions?

**Comment by H-CODE6** [comment]
Hello, after I run the code,  `threshold_values=True/False` is deprecated, please use `threshold=value` instead.
`fused_weight_gradient_mlp_cuda` module not found. gradient accumulation fusion with weight gradient computation disabled.
appears. I wonder if it has any impact?

**Comment by H-CODE6** [comment]
Hello, in training, the training data set is divided into Train dataset number of batch: 519
Val dataset number of batch: 130
Bench Test dataset number of batch: 130. What is the reason for this operation?

**Comment by H-CODE6** [comment]
I look forward to your answer very much

**Comment by himashi92** [comment]
"threshold_values=True/False is deprecated" is a warning, there is no impact from that for model training. In our preprint version in arxiv you can find all the implementation details including dataset splitting.

During model training, we divide the entire dataset into 3 sets: training (used for model training), validation (used for model validation and saved the best model) and test (evaluated on saved best model).

**Comment by H-CODE6** [comment]
OK, thank you very much. Do you still need to save some data in test.py, or just put all the data in the training set without dividing the test set.

**Comment by H-CODE6** [comment]
There is also fused_weight_gradient_mlp_cuda module not found. gradient accumulation fusion with weight gradient computation disabled.
appears. I wonder if it has any impact?

**Comment by himashi92** [comment]
No, you can use bench test dataset which is 130 scans in your case. In test.py https://github.com/himashi92/VT-UNet/blob/main/test.py#L102 use get_datastes instead of get_test_datasets. If you want to generate predictions for brats2021 validation dataset, then have a separate folder and use get_test_datasets method.

Regarding, "fused_weight_gradient_mlp_cuda module not found". This seems a torch version problem. Try updating torch. I am not sure whether that'll affect model training.

**Comment by H-CODE6** [comment]
OK, thank you very much for your reply. What version of torch do you use? Let me adjust my version.

**Comment by himashi92** [comment]
I am using pytorch=1.9.1 torchvision=0.10.1 and cuda=11.3

**Comment by H-CODE6** [comment]
OK, thank you very much. Did you use all the data in brats2021 in the experiment? Whether some data with bad marks are eliminated.

**Comment by himashi92** [comment]
yes, Brats2021 organizers updated the dataset.

**Comment by H-CODE6** [comment]
I'm sorry, I don't quite understand. Did you use all the data of brats2021, that is, 1251 data?

**Comment by himashi92** [comment]
we divide 1251 scans into 834, 208, 209 for training, validation and testing, respectively.

**Comment by H-CODE6** [comment]
OK, thank you very much for answering for me, and then express my thanks.

**Comment by H-CODE6** [comment]
Hello, because of the small version I use, you only provide the link of vt-unet-b pre training model. Can you provide the link of vt-unet-m and vt-unet-t pre training model?

**Comment by himashi92** [comment]
We'll release all the trained models with the new version of the code soon.

**Comment by H-CODE6** [comment]
Hello, I used the small version for training. If no pre training model is provided, put it into saved_ Model file, how can I test and verify the performance?

**Comment by himashi92** [comment]
You can train from scratch and evaluate the performance.

**Comment by H-CODE6** [comment]
Hello, do you mean to train the generated model_ best_ base. pth. Put tar into saved_ Under the model folder.

**Comment by H-CODE6** [comment]
Hello, I'd like to ask you something. Normally, you should not train first and then test with the trained model. I don't know why you are saving here_ Put the model provided in the model instead of the best model trained by yourself?

**Comment by H-CODE6** [comment]
Hello, when will the new version be released? I'm looking forward to it.

**Comment by himashi92** [comment]
At the moment we cannot provide you with trained best models for vt-unet T and S versions as the paper is still under review process. We'll update our repository soon. 

If you want to evaluate the VT-UNet model performance as given in the ReadME file, first train the model by yourself from scratch (That does not need our trained model weights, only pretrained weights of imagenet is needed. All the steps are given. Please follow instructions).  While training, the best model weights will be saved in the given respective path which is evaluated on val dataset. For example model_best.pth.tar.

This model_best.pth.tar can be then loaded using test.py and evaluated on the bench test dataset. 

We followed the conventional process of training a deep learning model like other researchers. Hope I addressed your concerns. For further details, go through the paper.

Thanks

**Comment by H-CODE6** [comment]
OK, thank you very much for your reply.

**Comment by H-CODE6** [comment]
Hello, what codes need to be modified if I want to use use bench test dataset which is 130 scans in your case. In test.py https://github.com/himashi92/VT-UNet/blob/main/test.py#L102 use get_datastes instead of get_test_datasets..

**Comment by H-CODE6** [comment]
We look forward to your reply.

**Comment by himashi92** [comment]
What do you mean? I cannot understand what you are saying. 

For training you can use train.py.
For testing during inference you can use test.py.

Feel free to modify those scripts as per your requirements. Everything is written in basic python and pytorch. If you don't understand the code and outcomes, try debugging and understand the code and how tensors work.

**Comment by H-CODE6** [comment]
I'm really sorry. Maybe I'm not accurate. Run test The data set tested in py needs to redefine a test set. In the code, it is get_ test_ Datasets, which I mentioned earlier, has been divided into a part as test. You said that these data can also be used in the test. Use get_ Datasets is enough, so I want to ask what code needs to be modified if you want to do so?

**Comment by H-CODE6** [comment]
I'm really sorry. Maybe I'm not accurate. Run test The data set tested in py needs to redefine a test set. In the code, it is get_ test_ Datasets. As I said earlier, a part has been divided into tests in the training. You said that these data can also be used in the test. Use get_ Datasets is enough, so I want to ask what code needs to be modified if you want to do this

**Comment by himashi92** [comment]
https://github.com/himashi92/VT-UNet/blob/2c5c5451a8e61d0884d5ddc3b487494602fd6e37/test.py#L102

change this line of code to following:

 _, _, bench_dataset = get_datasets(args.seed, fold_number=args.fold)

So that you can evaluate your 130 test set using test.py

Hope I answered your question

**Comment by H-CODE6** [comment]
OK, thank you very much for answering these questions for me. Thank you very much.

**Comment by himashi92** [comment]
No worries, good luck with your research!
